src/basic.jl

```diff
@@ -429,7 +429,7 @@ struct AddedOperator{T,
     function AddedOperator(ops)
         @assert !isempty(ops)
         _check_AddedOperator_sizes(ops)
-        T = promote_type(eltype.(ops)...)
+        T = mapreduce(eltype, promote_type, ops)
```

Comment:
make sure this doesn't allocate? Usually mapreduce is expensive.

Reply:
I had to change it because the broadcasting was actually making type instabilities.